File: PrioryxAI/constants/typography.ts
import { StyleSheet } from 'react-native';

export const Typography = StyleSheet.create({
  titleXL: { fontSize: 28, lineHeight: 34, fontWeight: '700', letterSpacing: -0.5 },
  titleLG: { fontSize: 24, lineHeight: 30, fontWeight: '700', letterSpacing: -0.3 },
  titleMD: { fontSize: 20, lineHeight: 26, fontWeight: '600', letterSpacing: -0.2 },
  titleSM: { fontSize: 17, lineHeight: 22, fontWeight: '600', letterSpacing: 0 },
  bodyLG: { fontSize: 17, lineHeight: 26, fontWeight: '400', letterSpacing: 0 },
  bodyMD: { fontSize: 15, lineHeight: 22, fontWeight: '400', letterSpacing: 0 },
  bodySM: { fontSize: 13, lineHeight: 18, fontWeight: '400', letterSpacing: 0 },
  captionLG: { fontSize: 12, lineHeight: 16, fontWeight: '500', letterSpacing: 0.2 },
  captionSM: { fontSize: 11, lineHeight: 14, fontWeight: '400', letterSpacing: 0.3 },
  overline: { fontSize: 11, lineHeight: 14, fontWeight: '500', letterSpacing: 1.2 },
  mono: { fontSize: 13, lineHeight: 18, letterSpacing: 0 },
});

File: PrioryxAI-iOS/PrioryxAI/DesignSystem/Typography.swift
import SwiftUI

extension Font {
  static let titleXL   = Font.system(size: 28, weight: .bold, design: .default)
  static let titleLG   = Font.system(size: 24, weight: .bold)
  static let titleMD   = Font.system(size: 20, weight: .semibold)
  static let titleSM   = Font.system(size: 17, weight: .semibold)
  static let bodyLG    = Font.system(size: 17, weight: .regular)
  static let bodyMD    = Font.system(size: 15, weight: .regular)
  static let bodySM    = Font.system(size: 13, weight: .regular)
  static let captionLG = Font.system(size: 12, weight: .medium)
  static let captionSM = Font.system(size: 11, weight: .regular)
  static let overline  = Font.system(size: 11, weight: .medium).uppercaseSmallCaps()
  static let mono      = Font.system(size: 13, design: .monospaced)
}
